templates/traefik.yaml

```diff
@@ -51,3 +51,8 @@ spec:
         defaultCN: traefik.localhost.localdomain
         defaultSANList:
           - "*"
+      forwardAuth:
+        entryPoints:
+        - https
+        address: http://traefik-forward-auth-kubeaddons.kubeaddons.svc.cluster.local:4181/
```

Comment:
Would traefik still work if we didn't enabled `dex` and all related dependencies ?

Reply:
No. This creates a dependency trail: traefik -> traefik-forward-auth -> dex. If any of these are unavailable, the Ops Portal is unusable.

What other paths do you think should be supported? Basic auth, no auth?

Reply:
Have you tried to use `ingress.kubernetes.io/auth-url` label on the Ingress object? I think we should only forward auth on known paths for `opsportal` and `dex-k8s-authenticator`, instead of forcing the same auth for all?

Reply:
cc @joejulian here as well

Reply:
I'll look into it. I was not aware that Traefik supported forward authentication on some paths only, because authentication occurs before any routing decisions are taken. I can't find any specifics in the Traefik docs, but https://stackoverflow.com/questions/50253357/add-basic-auth-for-admin-path-prefix suggests that it can be done (for basic auth at least). Note, currently, the rules provided in the `traefik-forward-auth` values spec define paths that are allowed without authentication.

Reply:
I just want to note that the change to use ingress labels will mean we default to allowing anonymous access, with deny for specific pages. It is usually better to default to denying access and then carefully allow access, as the current `traefik-forward-auth` whitelist does. This makes it more difficult to add a new path and mistakenly leave it open for access.

Reply:
:/. okay let me to check it too :(

Reply:
As discussed yesterday, we might initially go with this approach for now.

Reply:
I had a quick attempt at using `ingress.kubernetes.io/auth-url` but it did not work. So I'd like to merge as is and then look at the change separately, unless that is going to break anything now.